src/merlin.ml

```diff
@@ -5,29 +5,91 @@ open! No_io
 
 module SC = Super_context
 
+module Dropped_pp : sig
+  val warn
+    :  Loc.t
+    -> allow_approx_merlin:bool
+    -> dir:Path.t
+    -> reason:string
+    -> unit
+end = struct
+  module Input = struct
+    type t =
+      { loc : Loc.t
+      ; reason : string
+      ; dir : Path.t
+      }
+
+    let equal x y = Path.equal x.dir y.dir
+
+    let hash x = Path.hash x.dir
+
+    let to_sexp { dir; reason; loc }=
+      let open Dyn in
+      Record
+        [ "loc", Loc.to_dyn loc
+        ; "reason", String reason
+        ; "dir", Path.to_dyn dir
+        ]
+      |> Dyn.to_sexp
+  end
+
+  let warn_drop_pp_def =
```

Comment:
We don't need to memoize this. `Report_error` already takes care of not reporting the same error/warning twice.

Reply:
Whoops. I got rid of it.